C++/syntax_test_c.c

```diff
@@ -578,3 +578,52 @@ scanf("%ms %as %*[, ]", &buf);
 
 "foo % baz"
 /*   ^ - invalid */
+
+
+/////////////////////////////////////////////
+// Control Keywords
+/////////////////////////////////////////////
+
+if (x < 5)
+/* <- keyword.control */
+{}
+else
+/* <- keyword.control */
+{}
+
+switch (x)
+/* <- keyword.control */
+{
+case 1:
+/* <- keyword.control */
+    break;
+    /* <- keyword.control.flow.break */
+default:
+/* <- keyword.control */
+    break;
+    /* <- keyword.control.flow.break */
+}
+
+do
+/* <- keyword.control */
+{
+    if (y == 3)
+        continue;
+        /* <- keyword.control.flow.continue */
+} while (y < x);
+/*^ keyword.control */
+
+switch (a) {
+    case 1: break;
+/*        ^ punctuation.separator */
+    case 100 - 10: break;
+/*               ^ punctuation.separator */
+    default: break;
+/*         ^ punctuation.separator */
+}
```

Comment:
Could you please put these statements and expressions in a function definition (like `main`)? These added tests might screw up future tests because there's this global-local idea throughout the syntax definition. And strictly speaking, this is not valid C if it's at global scope.